packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -789,6 +789,41 @@ export function getMembers(target: IAnyStateTreeNode): IModelReflectionData {
     return reflected
 }
 
+export type CastedType<T> = T extends IStateTreeNode<infer STNC> ? STNC | T : T
+
+/**
+ * Casts a node snapshot or instance type to an instance type so it can be assigned to a type instance.
+ * Note that this is just a cast for the type system, this is, it won't actually convert a snapshot to an instance,
+ * but just fool typescript into thinking so.
+ * Casting only works on assignation operations, it won't work (compile) stand-alone.
+ * Technically it is not required for instances, but it is provided for consistency reasons.
+ *
+ * @example
+ * const ModelA = types.model({
+ *   n: types.number
+ * }).actions(self => ({
+ *   setN(aNumber: number) {
+ *     self.n = aNumber
+ *   }
+ * }))
+ *
+ * const ModelB = types.model({
+ *   innerModel: ModelA
+ * }).actions(self => ({
+ *   someAction() {
+ *     // this will allow the compiler to assign an snapshot to the property
+ *     self.innerModel = cast({ a: 5 })
+ *   }
+ * }))
+ *
+ * @export
+ * @param {CastedType<T>} snapshotOrInstance
+ * @returns {T}
+ */
+export function cast<T = never>(snapshotOrInstance: CastedType<T>): T {
```

Comment:
neat `never` trick!

Reply:
thanks! apparently TS will just use never when it doesn't have a return value to infer to